Userland/Services/WindowServer/Animation.h

```diff
@@ -33,18 +35,19 @@ class Animation : public RefCounted<Animation> {
     void set_duration(int duration_in_ms);
     int duration() const { return m_duration; }
 
-    bool update(Badge<Compositor>, Gfx::Painter&, Screen&, Gfx::DisjointIntRectSet& flush_rects);
+    bool update(Gfx::Painter&, Screen&, Gfx::DisjointIntRectSet& flush_rects);
 
     Function<void(float progress, Gfx::Painter&, Screen&, Gfx::DisjointIntRectSet& flush_rects)> on_update;
     Function<void()> on_stop;
 
 private:
-    Animation();
+    Animation() = default;
+
+    void call_stop_handler();
```

Comment:
could we make this public and take a `Badge<Compositor>` and make do without the `friend` that way?

Reply:
Sure.